frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseRebaseBranchOntoParentAction.java

```diff
@@ -227,6 +227,12 @@ public void run(ProgressIndicator indicator) {
                 "until ${gitRebaseParameters.getForkPointCommit().getHash()} commit " +
                 "onto ${gitRebaseParameters.getNewBaseBranch().getName()}");
 
+            /**
+             * Git4Idea ({@link git4idea.rebase.GitRebaseUtils#rebase}) does not allow to rebase in detached head state.
+             * However, it is possible with Git (performing checkout implicitly)
+             * and should be allowed in case of "Checkout and Rebase Onto Parent" Action.
+             * To pass the git4idea check in such a case we checkout the branch explicitly and then perform the actual rebase.
+             */
             if (shouldExplicitlyCheckout) {
               CheckoutSelectedBranchAction.doCheckout(
```

Comment:
Is this piece synchronous? So that we can be sure that once `GitRebaseUtils.rebase` is invoked, the checkout is already complete :thinking:

Reply:
yup, synchronous as something that is synchronous